KubeArmor/core/unorchestratedUpdates.go

```diff
@@ -429,12 +429,59 @@ func (dm *KubeArmorDaemon) ParseAndUpdateContainerSecurityPolicy(event tp.K8sKub
 	newPoint := tp.EndPoint{}
 
 	i := -1
+	renamedContainerIndex := -1
+	deletedPolicyIndex := -1
 
 	for idx, endPoint := range dm.EndPoints {
-		if kl.MatchIdentities(secPolicy.Spec.Selector.Identities, endPoint.Identities) {
+		if kl.MatchIdentities(secPolicy.Spec.Selector.Identities, endPoint.Identities) && i < 0 {
 			i = idx
 			newPoint = endPoint
-			break
+		}
+
+		for policyIndex, policy := range endPoint.SecurityPolicies {
+			if policy.Metadata["namespaceName"] == secPolicy.Metadata["namespaceName"] && policy.Metadata["policyName"] == secPolicy.Metadata["policyName"] && endPoint.EndPointName != containername {
+				renamedContainerIndex = idx
+				deletedPolicyIndex = policyIndex
```

Comment:
Today KubeArmor only supports, single container policies. But in future there will be Policy Matching to suppport policies on multiple containers. This logic will not scale when we add that. We should implement this in such a way, that it's not a special case handling but handle it in a natural way so as to address the issue. WDYT?